src/shared/clients/jit.py

```diff
@@ -150,9 +155,54 @@ def add_teams_to_asset(token, asset: Asset, teams: List[str]):
         }
         response = requests.patch(url, json=payload, headers=headers)
         if response.status_code == 200:
-            logger.info(f"Team(s) synced to asset '{asset.asset_name}' successfully.")
+            logger.info(
+                f"Team(s) synced to asset '{asset.asset_name}' successfully.")
         else:
             logger.error(f"Failed to add teams to asset '{asset.asset_id}'. Status code: "
                          f"{response.status_code}, {response.text}")
     except Exception as e:
         logger.error(f"Failed to add teams to asset: {str(e)}")
+
+
+def _perform_set_manual_team_members(token: str, team_id: str,
+                                     members: List[str], team_name: str) -> Optional[List[str]]:
+    try:
+        url = f"{get_jit_endpoint_base_url()}/teams/{team_id}/members"
+        headers = get_request_headers(token)
+        payload = {
+            "members": members
+        }
+        response = requests.put(url, json=payload, headers=headers)
+        if response.status_code == 200:
+            failed_members = response.json().get("failed_members", [])
+            if failed_members:
+                logger.warning(
+                    f"Failed to set some members for team '{team_name}' with ID '{team_id}': {failed_members}")
+            else:
+                logger.info(
+                    f"Members set for team '{team_name}' with ID '{team_id}' successfully.")
+            return failed_members
+        else:
+            logger.error(f"Failed to set members for team '{team_name}' with ID '{team_id}'. Status code: "
+                         f"{response.status_code}, {response.text}")
+            return None
+    except Exception as e:
+        logger.error(
+            f"Failed to set members for team with ID '{team_id}': {str(e)}")
+        return None
+
+
+def set_manual_team_members(token: str, team_id: str, members: List[str], team_name: str) -> None:
+    retry_count = 0
+    failed_members = _perform_set_manual_team_members(
+        token, team_id, members, team_name)
+    while retry_count <= MAX_RETRIES and failed_members:
```

Comment:
so just for clarification - we only retry in cases where the api returned 200 and failed members?

Reply:
Yes. Right now if any other error comes up, we log it and move on. 

Just to be safe I added a general try-except in the `process_members` func